src/seaport.ts

```diff
@@ -136,15 +140,23 @@ export class Seaport {
   }
 
   private _getSigner(accountAddress?: string): Signer {
-    if (this.signer) {
-      return this.signer;
-    }
-
     if (!(this.provider instanceof providers.JsonRpcProvider)) {
       throw new Error("Either signer or a JsonRpcProvider must be provided");
     }
 
-    return this.provider.getSigner(accountAddress);
+    /**
+     * If accountAddress is passed in explicitly, we should use that as the signer
+     */
```

Comment:
Don't put docstrings in the middle of functions

/** ... */ is a docstring
/* ... */ is a block comment

Stylistically I think there's no need for any kind of block comment inside a function, `//` comments should be fine and I think are easier for IDEs to toggle on-off

Also style: don't put an extra newline between a comment and the code that it refers to